hal/crc_api.h

```diff
@@ -35,6 +35,19 @@ typedef enum crc_polynomial {
     POLY_32BIT_ANSI  = 0x04C11DB7, // x32+x26+x23+x22+x16+x12+x11+x10+x8+x7+x5+x4+x2+x+1
 } crc_polynomial_t;
 
+typedef struct crc_mbed_config {
```

Comment:
I see that conception has changed. This change has massive impact on my test, so please provide requirements for testing.
I have also few questions, so I can start re-factoring my test.
Do we assume now that CRC platforms must support all of the listed options?
Why CRC width is not specified in `crc_mbed_config` (we can have different CRC width for the same polynomial)?
Does conception with predefined typical polynomials is eventually abandoned?

Reply:
> Do we assume now that CRC platforms must support all of the listed options?

I think yes? Some of them can be handled manually like final_xor by writing it to the module before returning the value, but I believe reflect_in/out/initial_seed should be supported as options.

> Why CRC width is not specified in crc_mbed_config (we can have different CRC width for the same polynomial)?

As far as I can tell the width of the CRC polynomial can be derived from the polynomial that's passed into the module, the largest polynomial that is set defines the width, at least for all the ones I've seen. I suppose an additional check needs to be added to the K64F implementation to ensure that either bit-31 is set indicating 32-bit polynomial, or that bit-15 is the highest bit set indicating a 16-bit one.

> Does conception with predefined typical polynomials is eventually abandoned?

Yeah, it seemed to be more of an indicator for which polynomials have table support.

Reply:
Thanks for the answers. I have two more questions:
1. Do we still limit the number of supported polynomials to ones provided in `crc_polynomial_t` enum with additional config options (initial_xor, final_xor, reflect_in, reflect_out).
2. Why initial value config option is not available?

Reply:
> Do we still limit the number of supported polynomials to ones provided in crc_polynomial_t enum with additional config options (initial_xor, final_xor, reflect_in, reflect_out).

No, any polynomial is now supported if the hardware allows it.

> Why initial value config option is not available?

This is the same as an initial_xor? Most platforms that defines that option just uses it as an initial value in the hardware.

Reply:
> This is the same as an initial_xor?
Looks like it is :-) Thanks.